diff --git a/components/network_session_configurator/common/network_switch_list.h b/components/network_session_configurator/common/network_switch_list.h
index 4f2524d7b9..034bb25bd9 100644
--- a/components/network_session_configurator/common/network_switch_list.h
+++ b/components/network_session_configurator/common/network_switch_list.h
@@ -76,3 +76,6 @@ NETWORK_SWITCH(kHttp2GreaseFrameType, "http2-grease-frame-type")
 // afterwards.  Only affects HTTP/2 request streams, not proxy or bidirectional
 // streams.
 NETWORK_SWITCH(kHttp2EndStreamWithDataFrame, "http2-end-stream-with-data-frame")
+
+// CUSTOM: override PrivacyMode values with PRIVACY_MODE_DISABLED
+NETWORK_SWITCH(kIgnorePrivacyMode, "ignore-privacy-mode")
diff --git a/net/url_request/url_request.cc b/net/url_request/url_request.cc
index 9138aae019..b8cdd246a5 100644
--- a/net/url_request/url_request.cc
+++ b/net/url_request/url_request.cc
@@ -9,6 +9,7 @@
 #include "base/bind.h"
 #include "base/bind_helpers.h"
 #include "base/callback.h"
+#include "base/command_line.h"
 #include "base/compiler_specific.h"
 #include "base/metrics/histogram_macros.h"
 #include "base/rand_util.h"
@@ -38,6 +39,7 @@
 #include "net/url_request/url_request_job_factory.h"
 #include "net/url_request/url_request_netlog_params.h"
 #include "net/url_request/url_request_redirect_job.h"
+#include "components/network_session_configurator/common/network_switches.h"
 #include "url/gurl.h"
 #include "url/origin.h"
 
@@ -1021,6 +1023,12 @@ bool URLRequest::CanSetCookie(const net::CanonicalCookie& cookie,
 }
 
 PrivacyMode URLRequest::DeterminePrivacyMode() const {
+  const bool ignore_privacy =
+     base::CommandLine::ForCurrentProcess()
+        ->HasSwitch(::switches::kIgnorePrivacyMode);
+  if (ignore_privacy)
+    return PRIVACY_MODE_DISABLED;
+
   if (!allow_credentials_) {
     // |allow_credentials_| implies LOAD_DO_NOT_SAVE_COOKIES.
     DCHECK(load_flags_ & LOAD_DO_NOT_SAVE_COOKIES);
diff --git a/services/network/network_context.cc b/services/network/network_context.cc
index 3773583bed..663a2a43ab 100644
--- a/services/network/network_context.cc
+++ b/services/network/network_context.cc
@@ -1622,6 +1622,12 @@ void NetworkContext::PreconnectSockets(
   }
   request_info.network_isolation_key = network_isolation_key;
 
+  const bool ignore_privacy =
+     base::CommandLine::ForCurrentProcess()
+        ->HasSwitch(::switches::kIgnorePrivacyMode);
+  if (ignore_privacy)
+    request_info.privacy_mode = net::PRIVACY_MODE_DISABLED;
+
   net::HttpTransactionFactory* factory =
       url_request_context_->http_transaction_factory();
   net::HttpNetworkSession* session = factory->GetSession();
